ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -338,4 +377,34 @@ private void addStatTask(ASTNode root, Table table, Path oldPartitionLocation, P
       moveTask.addDependentTask(statTask);
     }
   }
+
+  public StringBuilder constructDeleteQuery(String qualifiedTableName, Table table, Map<String, String> partitionSpec)
+      throws SemanticException {
+    StringBuilder sb = new StringBuilder().append("delete from ").append(qualifiedTableName)
+            .append(" where ");
+    boolean first = true;
+    for (String key : partitionSpec.keySet()) {
+      ColumnInfo columnInfo = table.getStorageHandler().getColumnType(table, key);
+      if (columnInfo.getObjectInspector() instanceof PrimitiveObjectInspector) {
+        PrimitiveObjectInspector primitiveObjInspector = (PrimitiveObjectInspector) columnInfo.getObjectInspector();
+        String value = partitionSpec.get(key);
+        boolean shouldEncloseQuotes = TypeInfoUtils.shouldEncloseQuotes(primitiveObjInspector.getPrimitiveCategory());
+        if (value != null) {
+          sb.append(first ? "" : " and ").append(HiveUtils.unparseIdentifier(key, conf)).append(" = ");
+          if (shouldEncloseQuotes) {
```

Comment:
should we extract this quote logic into some util function?

Reply:
Done.